packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -134,6 +134,12 @@ abstract class Target {
   /// A list of zero or more depfiles, located directly under {BUILD_DIR}.
   List<String> get depfiles => const <String>[];
 
+  /// Whether this target can be executed with the given [environment].
+  ///
+  /// Returning `true` will cause [build] to be skipped. This is equivalent
+  /// to a build that produces no outputs.
+  bool canSkip(Environment environment) => false;
```

Comment:
Using this we can safely "skip" the target if l10n.yaml does not exist and clean up the old outputs.